lib/mint/types.ex

```diff
@@ -3,6 +3,12 @@ defmodule Mint.Types do
   HTTP-related types.
   """
 
+  @typedoc """
+  A hostname, ip address, unix domain socket path, :loopback, or any
+  other term representing an internet address.
+  """
+  @type address() :: :inet.socket_address() | :inet.hostname()
```

Comment:
```suggestion
  @type address() :: :inet.socket_address() | String.t()
```

Related to https://github.com/elixir-mint/mint/pull/283#discussion_r518203488 

Also I misunderstood the erlang `any` to be the same as the elixir typespec `any` and I just realized it would actually be the atom `:any` so I was wrong with my most recent comment in the previous discussion 😓 sorry about that. I will say going back and forth from erlang to elixir every day the past couple weeks has been tough 😅 

/cc @whatyouhide and @ericmj